src/main/java/com/google/sps/servlets/EventsServlet.java

```diff
@@ -0,0 +1,199 @@
+    );
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(10);
+    Query query = new Query(EVENT).addSort(TIMESTAMP, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery pq = datastore.prepare(query);
+
+    QueryResultList<Entity> results = pq.asQueryResultList(fetchOptions);
+
+    List<Event> events = new ArrayList<>();
+    for (Entity entity : results) {
+        long timestamp = (long) entity.getProperty(TIMESTAMP);
+        String summary = (String) entity.getProperty(SUMMARY);
+        String description = (String) entity.getProperty(DESCRIPTION);
+        String location = (String) entity.getProperty(LOCATION);
+        Date startTime = (Date) entity.getProperty(START_TIME);
+        Date endTime = (Date) entity.getProperty(END_TIME);
+        String category = (String) entity.getProperty(CATEGORY);
+
+        DateTime startDateTime = new DateTime(startTime);
+        EventDateTime start = new EventDateTime()
+            .setDateTime(startDateTime)
+            .setTimeZone(UTC_TIMEZONE);
+        DateTime endDateTime = new DateTime(endTime);
+        EventDateTime end = new EventDateTime()
+            .setDateTime(endDateTime)
+            .setTimeZone(UTC_TIMEZONE);
+        Event event = new Event()
+            .setSummary(summary)
+            .setLocation(location)
+            .setDescription(description)
+            .setStart(start)
+            .setEnd(end);
+
+        ExtendedProperties ep = new ExtendedProperties();
+        ep.set(CATEGORY, category);
+        event.setExtendedProperties(ep);
+    
+        events.add(event);
+    }
+
+    response.setContentType("application/json; charset=UTF-8");
+    response.setCharacterEncoding("UTF-8");
+
+    String json = convertToJson(events);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String eventSummary = request.getParameter(SUMMARY);
+      String eventDescription = request.getParameter(DESCRIPTION);
+      String eventLocation = request.getParameter(LOCATION);
+      String eventDateString = request.getParameter(DATE);
+      String eventStartTimeString = request.getParameter(START_TIME);
+      String eventEndTimeString = request.getParameter(END_TIME);
+      String timezoneOffset = request.getParameter(USER_TIMEZONE);
+      String category = request.getParameter(CATEGORY);
+
+      if (!CATEGORIES.contains(category)) category = "other";
+  
+      Date eventStartDateTime = getEventDateTime(eventDateString, eventStartTimeString, timezoneOffset);
+      Date eventEndDateTime = getEventDateTime(eventDateString, eventEndTimeString, timezoneOffset);
+      if (eventStartDateTime == null || eventEndDateTime == null) return;
```

Comment:
Validate eventStartDateTime < eventEndDateTime, and if not decide on how to handle it. One way to handle it would be to use the earliest time as the start.